docs/functionalabstractions/concretetypes/ord.md

```diff
@@ -3,4 +3,190 @@ id: ord
+```
+
+We can also use it to compare two business accounts or two consumer accounts.
+
+```scala mdoc
+val consumerAccount1: ConsumerAccount = ConsumerAccount("abc")
+val consumerAccount2: ConsumerAccount = ConsumerAccount("def")
+
+val compareConsumerAccounts: Boolean =
+  consumerAccount1 < consumerAccount2 // okay
+```
+
+This just works because a consumer account is a customer account and we know how to order customer accounts.
+
+Let's see what happens when we try to do the same thing with the `Ordering` type class defined in the Scala standard library.
+
+```scala mdoc:reset
+import scala.math.Ordering.Implicits._
+
+sealed trait CustomerAccount
+
+case class BusinessAccount(id: String) extends CustomerAccount
+case class ConsumerAccount(id: String) extends CustomerAccount
+
+implicit val CustomerAccountScalaOrdering: scala.math.Ordering[CustomerAccount] =
+  new scala.math.Ordering[CustomerAccount] {
+    def compare(left: CustomerAccount, right: CustomerAccount): Int =
+      (left, right) match {
+        case (BusinessAccount(id1), BusinessAccount(id2)) => id1 compare id2
+        case (BusinessAccount(_), ConsumerAccount(_))     => -1
+        case (ConsumerAccount(_), BusinessAccount(_))     => 1
+        case (ConsumerAccount(id1), ConsumerAccount(id2)) => id1 compare id2
+      }
+}
+```
+
+Besides the lack of specific types in modeling the result of the compare operator this works fine for comparing two customer accounts.
+
+```scala mdoc
+val customerAccount1: CustomerAccount = BusinessAccount("abc")
+val customerAccount2: CustomerAccount = ConsumerAccount("def")
+
+val compareCustomerAccounts: Boolean =
+  customerAccount1 < customerAccount2 // okay
+```
+
+But look what happens when we try to compare two consumer accounts.
+
+```scala mdoc:fail
+val consumerAccount1: ConsumerAccount = ConsumerAccount("abc")
+val consumerAccount2: ConsumerAccount = ConsumerAccount("def")
+
+val compareConsumerAccounts: Boolean =
+  consumerAccount1 < consumerAccount2 // does not compile
+```
+
+This code doesn't compile! The Scala compiler is unable to compare the consumer accounts because of the lack of variance on `scala.math.Ordering`.
+
+We could widen the type of `ConsumerAccount` to `CustomerAccount` but that is unergonomic. And it wouldn't even work if, for example, we were trying to sort the list and needed to maintain the type information that these were consumer accounts rather than just customer accounts.
+
+The `Ord` abstraction helps us avoid problems like this and integrates seamlessly with the rest of ZIO Prelude.
+
+## Defining Ord Instances
+
+`Ord` instances are already defined for all the types in `ZIO` and the Scala standard library that have a total ordering.
+
+ZIO Prelude is also very good at automatically deriving `Ord` instances for more complex data types like tuples. For example if we have a pair of `Double` values representing x and y coordinates an ordering is already automatically defined for it.
```

Comment:
Is not it rather Scalac that is responsible for automatic derivation or there's Prelude doing something specific in terms of typeclass derivation here?

Reply:
General you may back-tick `x` and `y` throughout this file.

Reply:
The Scala compiler is doing the derivation but it is based on the "axioms" we give it in terms of implicit instances for primitive types and implicit definitions to derive instances for more complex types.